Client/qtTeamTalk/moveusersdlg.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2023, Bjørn D. Rasmussen, BearWare.dk
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "moveusersdlg.h"
+#include "common.h"
+#include "appinfo.h"
+
+extern TTInstance* ttInst;
+
+MoveUsersDlg::MoveUsersDlg(const QVector<int>& userIds, const QVector<int>& channelIds, QWidget *parent)
+    : QDialog(parent), userIds(userIds), channelIds(channelIds)
+{
+    setupUi();
+    populateUsers();
+    populateChannels();
+}
+
+void MoveUsersDlg::setupUi()
+{
+    setWindowTitle(tr("Move Users"));
+    setWindowIcon(QIcon(APPICON));
+
+    QVBoxLayout* mainLayout = new QVBoxLayout(this);
+
+    QGroupBox* usersGroup = new QGroupBox(tr("Users"), this);
+    QVBoxLayout* usersLayout = new QVBoxLayout(usersGroup);
```

Comment:
Clang-Tidy says "usersLayout" may memory leak. I just can't figure out why. Can you? It appears to be the case if userIds.size() is 0. However, if I put a check for userIds.size()>0 then it still complains :(

Reply:
Maybe just add a QLabel if userids is empty to indicates there is no user available. This QLabel will never be visible but maybe clang-tidy will be happy. Not sure :)

Reply:
Hm, that also didn't work (with the QLabel). I'll merge. It looks like a false-positive.